File: Claude/3.Desarrollo Web Endpoint RESTful/c++/endpoint.cpp
#include "crow.h"
#include <string>
#include <nlohmann/json.hpp>

using json = nlohmann::json;

int main()
{
    crow::SimpleApp app;

    // GET endpoint
    CROW_ROUTE(app, "/saludo").methods("GET"_method)
    ([](const crow::request& req) {
        std::string nombre = req.url_params.get("nombre") ? 
                           req.url_params.get("nombre") : "Usuario";
        
        json response;
        response["mensaje"] = "Hola, " + nombre;
        response["status"] = "success";
        
        return crow::response(200, response.dump());
    });

    // POST endpoint
    CROW_ROUTE(app, "/saludo").methods("POST"_method)
    ([](const crow::request& req) {
        std::string nombre = "Usuario";
        
        try {
            json request_body = json::parse(req.body);
            if (request_body.contains("nombre")) {
                nombre = request_body["nombre"];
            }
        } catch (const std::exception& e) {
            // Si hay error en JSON, usar nombre por defecto
        }
        
        json response;
        response["mensaje"] = "Hola, " + nombre;
        response["status"] = "success";
        
        return crow::response(201, response.dump());
    });

    // Configurar CORS y headers
    app.get_middleware<crow::CORSHandler>().global()
        .headers("Content-Type", "Authorization")
        .methods("GET"_method, "POST"_method);

    app.port(8080).multithreaded().run();
    return 0;
}

File: Claude/3.Desarrollo Web Endpoint RESTful/python/endpoint.py
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route('/saludo', methods=['GET'])
def obtener_saludo():
    nombre = request.args.get('nombre', 'Usuario')
    return jsonify({
        'mensaje': f'Hola, {nombre}',
        'status': 'success'
    })

@app.route('/saludo', methods=['POST'])
def crear_saludo():
    data = request.get_json()
    nombre = data.get('nombre', 'Usuario') if data else 'Usuario'
    return jsonify({
        'mensaje': f'Hola, {nombre}',
        'status': 'success'
    }), 201

if __name__ == '__main__':
    app.run(debug=True, host='0.0.0.0', port=5000)